### DISCUSSION TITLE
How long are event logs stored?

### DISCUSSION THREAD

**Discussion by Laderlappen** [zkSync-Community-Hub/zksync-developers#107] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement.

Good day!

We have a blog type dapp on Ethereum. We want to deploy it on ZKSync era too. We store the text of a blog post in a emited event's data. To view the history of blog posts made in the past we simply scan the transaction logs for events (with getLogs rpc request) and look for posts.

Hence it is important that we can get posts even if they were posted a long time ago to the chain.
So how long is log data (For calls and events) stored on a node and are they accessable? 
Would it work the same as on Ethereum L1?

### What is the specific issue or error you're encountering?

No issue yet. Just a question.

### Can you share the error messages or logs you're receiving, if any?

N/A

### Have you made any recent changes to the contract before encountering this issue?

N/A

### Are there any external libraries or contracts that your contract interacts with?

N/A

### Can you provide the relevant portions of your contract code where the issue is occurring?

N/A

### Have you tried to isolate the problem, and if so, what were the results?

N/A

### What steps have you already taken to try to resolve the issue?

Read the docs.

**Comment by RomanBrodetski** [comment, marked as answer]
Hello @Laderlappen ,

Currently we store the events indefinitely and make them available in the API. We might change our approach to saving them, but it's unlikely we'll ever completely drop old events - there will be a way to retrieve them.

**Reply by Laderlappen** [reply]
Hi,

Ok great. Thank you for the answer.
